packages/firestore/src/util/misc.ts

```diff
@@ -16,20 +16,26 @@
  */
 
 import { assert } from './assert';
+import { Platform } from '../platform/platform';
 
 export type EventHandler<E> = (value: E) => void;
 export interface Indexable {
   [k: string]: unknown;
 }
 
 export class AutoId {
-  static newId(): string {
+  static newId(platform: Platform): string {
     // Alphanumeric characters
     const chars =
       'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
     let autoId = '';
     for (let i = 0; i < 20; i++) {
-      autoId += chars.charAt(Math.floor(Math.random() * chars.length));
+      // Length of `chars` is 62. We generate random values between 0 and 247.
```

Comment:
Mark this explicitly inclusive. Usually "between" implies one or both end points are excluded from the set. That is, use "Random values between 0 and 247 (inclusive)."

Reply:
Done.